### PR TITLE
Protocol fee collection (ETH-572)

### PR DESCRIPTION
**Pull request by jtakalai**

### CODE REVIEW

**Comment by jtakalai** [comment]
Moved the renaming into a separate PR, though the "functional breaking change" (added return value) was already in this PR.

**Review by samt1803 — APPROVED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/Operator.sol [hunk 1]
```diff
@@ -438,6 +422,27 @@ contract Operator is Initializable, ERC2771ContextUpgradeable, IERC677Receiver,
...
+        uint rewardLimit
+    ) {
+        sponsorshipAddresses = new address[](sponsorships.length);
+        earnings = new uint[](sponsorships.length);
+        for (uint i = 0; i < sponsorships.length; i++) {
+            Sponsorship sponsorship = sponsorships[i];
+            sponsorshipAddresses[i] = address(sponsorship);
+            earnings[i] = sponsorship.getEarnings(address(this));
+        }
+        rewardLimit = getApproximatePoolValue() * streamrConfig.poolValueDriftLimitFraction() / 1 ether;
```
Comment (samigabor) [inline packages/network-contracts/contracts/OperatorTokenomics/Operator.sol:443]: Reward limit makes me thing there is a limit for the sponsorship earnings. Would driftLimit be more accurate and in sync with the streamr config namings?

File: packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol [hunk 2]
```diff
@@ -166,6 +172,10 @@ contract StreamrConfig is Initializable, UUPSUpgradeable, AccessControlUpgradeab
         poolValueDriftLimitFraction = 0.05 ether;
         poolValueDriftPenaltyFraction = 0.5 ether;
 
+        // protocol fee
+        protocolFeeFraction = 0.05 ether;
+        protocolFeeBeneficiary = address(0xD20b71E226f1a63794e40f02c7fD511E798F17Aa);
```
Comment (samigabor) [inline packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol:177]: Should we pass the beneficiary as a constructor param? Or add a setter for it?

**Review by samigabor — APPROVED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/Operator.sol (hunk 1 above)
Comment (jtakalai) [reply packages/network-contracts/contracts/OperatorTokenomics/Operator.sol:443]: it's going to be changed in the other PR  -___-

File: packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol (hunk 2 above)
Comment (jtakalai) [reply packages/network-contracts/contracts/OperatorTokenomics/StreamrConfig.sol:177]: Good idea actually! Also I'll add setters.

**Pull request merged**